### DISCUSSION TITLE
Some questions about Dory setup

### DISCUSSION THREAD

**Discussion by Lengxiaoyi** [spaceandtimefdn/sxt-proof-of-sql#395] [General]
# Background and Motivation
Hi, when I run the example example using only the CPU, I find that when I set const DORY_SETUP_MAX_NU: usize = 13, the time will only consume roughly 10s, and when I set it to 15, the time consumes 49s, and the performance of the CPU I'm using is pretty much the same as described in the code, but when I use the blitzar feature I find that the time transformations can increase by 2 squares with the DORY_SETUP_MAX_NU number setting, which is not GPU enabled in the dory setting. The time transform can increase by 2 squares with the number setting of DORY_SETUP_MAX_NU, the GPU was not enabled at the time of the dory setting I can't pinpoint the cause of the problem need your help, good luck!
# Changes Required
Please help me understand this.

**Comment by JayWhite2357** [comment]
Hey @Lengxiaoyi,
The reason that the setup takes much longer when the GPU is enabled is that `blitzar` does a one-time precomputation in order for the GPU to be more efficient. This precomputation is very large, but can be saved and later loaded from a file. The reason that the CPU is so much faster is that it doesn't do this precomputation because it has `blitzar` disabled.

We have published a few of these parameters here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/releases/tag/dory-prover-params-nu-16 (or https://github.com/spaceandtimelabs/sxt-proof-of-sql/releases/tag/max-nu-14). But, you can do it youself.

Take a look at this: https://github.com/spaceandtimelabs/sxt-proof-of-sql/tree/main/crates/proof-of-sql/utils/generate-parameters.

**Reply by Lengxiaoyi** [reply]
Thank you very much for your reply, I didn't notice that part and now I can understand it.XD～
